lib/module.js

```
@@ -513,14 +513,10 @@ Module._load = function(request, parent, isMain) {
 };
 
 function tryModuleLoad(module, filename) {
-  var threw = true;
   try {
     module.load(filename);
-    threw = false;
-  } finally {
-    if (threw) {
-      delete Module._cache[filename];
-    }
+  } catch (err) {
+    delete Module._cache[filename];
```

This silently suppresses the error. You need to `throw err`.